tests/utils/network_setup.go

```diff
@@ -0,0 +1,270 @@
+package utils
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+	"os"
+	"time"
+
+	"github.com/ava-labs/avalanche-network-runner/rpcpb"
+	"github.com/ava-labs/avalanchego/ids"
+	relayerEvm "github.com/ava-labs/awm-relayer/vms/evm"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/ethclient"
+	"github.com/ava-labs/subnet-evm/plugin/evm"
+	"github.com/ava-labs/subnet-evm/rpc"
+	"github.com/ava-labs/subnet-evm/tests/utils/runner"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	FundedKeyStr = "56289e99c94b6912bfc12adc093c9b51124f0dc54ac7a766b2bc5ccf558d8027"
+)
+
+var (
+	TeleporterContractAddress        common.Address
+	SubnetIDs                        []ids.ID
+	SubnetA, SubnetB                 ids.ID
+	BlockchainIDA, BlockchainIDB     ids.ID
+	ChainANodeURIs, ChainBNodeURIs   []string
+	FundedAddress                    = common.HexToAddress("0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC")
+	FundedKey                        *ecdsa.PrivateKey
+	ChainAWSClient, ChainBWSClient   ethclient.Client
+	ChainARPCClient, ChainBRPCClient ethclient.Client
+	ChainARPCURI, ChainBRPCURI       string
+	ChainAIDInt, ChainBIDInt         *big.Int
```

Comment:
We should be very careful with global exported values like this. We should at the very least leave a comment here detailing when they're supposed to be initialized, and that they should not be modified.

Reply:
That's a good call. I think a `GetSubnetInfo` exported function would be more appropriate, that way these globals are not modifiable outside of `package utils`.

Reply:
Put all these vars behind exported getters. Each test case should extract these at the beginning of the test.